pkg/restore/restore.go

```diff
@@ -455,11 +619,36 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 			if !existingNamespaces.Has(mappedNsName) {
 				logger := ctx.log.WithField("namespace", nsName)
 				ns := getNamespace(logger, filepath.Join(dir, api.ResourcesDir, "namespaces", api.ClusterScopedDir, nsName+".json"), mappedNsName)
-				if _, err := kube.EnsureNamespaceExists(ns, ctx.namespaceClient); err != nil {
+				nsIsReady, err := kube.EnsureNamespaceExistsAndIsReady(ns, ctx.namespaceClient)
+				if err != nil {
 					addArkError(&errs, err)
 					continue
 				}
 
+				if !nsIsReady {
+					tempNs, err := ctx.namespaceClient.Get(mappedNsName, metav1.GetOptions{})
+					if err != nil && apierrors.IsNotFound(err) {
+						addArkError(&errs, err)
+						continue
+					}
+
+					if isNSTerminating(tempNs) {
+						logger.Infof("Namespace still terminating, waiting")
+
+						ctx.nsWaiter.addDeleteChan(mappedNsName)
+						_, err = ctx.nsWaiter.waitForDelete(mappedNsName, ctx.namespaceTimeout)
```

Comment:
I think we probably need to check again here to see if it's been terminated, since it's possible it terminated between our first check and adding the delete channel for it.  Only if it still hasn't been terminated to we then need to wait for delete